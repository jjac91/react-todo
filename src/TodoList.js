import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuid } from "uuid";

const TodoList = () => {
  const [todos, setTodos] = useState([]);

  const addTodo = (todoData) => {
    let newTodo = { ...todoData, id: uuid() };
    setTodos((todos) => [...todos, newTodo]);
  };

  const remove = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  const renderTodos = todos.map((todo) => (
    <Todo id={todo.id} todo={todo.todo} handleRemove={remove} key={todo.id} />
  ));

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {renderTodos}
    </div>
  );
};

export default TodoList;
