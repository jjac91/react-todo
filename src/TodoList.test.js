import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";


function addTodo(todoList, todo = "eat") {
  const todoInput = todoList.getByLabelText("Add Todo:");
  fireEvent.change(todoInput, { target: { value: todo } });
  const button = todoList.getByText("Add Todo");
  fireEvent.click(button);
}
it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a todo", function () {
  const todoList = render(<TodoList />);
  expect(todoList.queryByText("X")).not.toBeInTheDocument();

  addTodo(todoList);
  const deleteButton = todoList.getByText("X");
  expect(deleteButton).toBeInTheDocument();
  expect("eat").toBeInTheDocument;
});

it("can remove a box", function () {
  const todoList = render(<TodoList />);
  addTodo(todoList);
  const deleteButton = todoList.getByText("X");
  fireEvent.click(deleteButton);
  expect(todoList.queryByText("X")).not.toBeInTheDocument();
});
