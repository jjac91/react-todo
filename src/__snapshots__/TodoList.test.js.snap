// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        Add Todo:
      </label>
      <input
        id="todo"
        name="todo"
        type="text"
        value=""
      />
      <button
        id="newToodoButton"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
